src/main/java/de/bsi/secvisogram/csaf_cms_backend/couchdb/CouchDBFilterCreator.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Selectors that mark an array field in the JSON structure
+     */
+    private final List<String[]> arrayFieldSelectors;
+
+    public CouchDBFilterCreator() {
+
+        this.arrayFieldSelectors = Collections.emptyList();
+
+    }
+
+    public CouchDBFilterCreator(List<String[]> arrayFieldSelectors) {
+
+        this.arrayFieldSelectors = Collections.unmodifiableList(arrayFieldSelectors);
+    }
+
+    /**
+     * Create Cloudant expression for given expression
+     * @param expr the expression to convert
+     * @return the converted expression object
+     */
+    public Map<String, Object> expression2CouchDbFilter(Expression expr) {
+
+        return expression2Filter(expr);
+    }
+
+    /**
+     * Create recursive Cloudant expression for given expression
+     * @param expr the expression to convert
+     * @return the converted expression object
+     */
+    private Map<String, Object> expression2Filter(Expression expr) {
+
+        ExpressionHandler<Map<String, Object>> couchDbExprHandler = new ExpressionHandler<>() {
+            @Override
+            public Map<String, Object> and(AndExpression andExpr) {
+                List<Object> andExpressions = andExpr.getExpressions()
+                        .stream()
+                        .map(expr -> expression2Filter(expr))
+                        .collect(Collectors.toList());
+
+                return createAndExpression(andExpressions);
+            }
+
+            @Override
+            public Map<String, Object> operator(OperatorExpression opExpr) {
+                return createOperatorExpression(opExpr);
+            }
+        };
+        return expr.handleExpr(couchDbExprHandler);
+    }
+
+    /**
+     * Create Cloudant expression for given operator expression, like
+     * <pre>
+     *     "document": {
+     *             "acknowledgments": {
+     *                "$elemMatch": {
+     *                   "urls": {
+     *                      "$elemMatch": {
+     *                         "$eq": "exccellent.de"
+     *                      }
+     *                   }
+     *                }
+     *             }
+     *          }
+     * </pre>
+     * @param opExpr the expression to convert
+     * @return the converted expression object
+     */
+    private Map<String, Object> createOperatorExpression(OperatorExpression opExpr) {
+
+        final String couchDBOperator = convertOperator(opExpr);
+
+        final Object compareValue;
+        if (opExpr.getValueType() == TypeOfValue.Decimal) {
+            compareValue = Double.valueOf(opExpr.getValue());
+        } else if (opExpr.getValueType() == TypeOfValue.Boolean) {
+            compareValue = "true".equals(opExpr.getValue());
+        } else if (opExpr.getOperatorType() == TypeOfOperator.ContainsIgnoreCase) {
+            compareValue = "^.*(?i)" + opExpr.getValue() + "(?i).*";
```

Comment:
the (?i) ist to get the opExpr.getValue() insensitive

Reply:
As discussed, I misinterpreted the intention here. Checking [the modifier documentation](https://www.regular-expressions.info/modifiers.html) again, you'd need to use `?-i` the second time to only wrap a part of the regex as case insensitive. However, since the `.`s match anything / are case insensitive anyway, starting the regex with a single `?i` would be enough.

Reply:
Ok